ee/tables/jwt/table.go

```diff
@@ -0,0 +1,177 @@
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/kolide/launcher/ee/dataflatten"
+	"github.com/kolide/launcher/ee/tables/dataflattentable"
+	"github.com/kolide/launcher/ee/tables/tablehelpers"
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+// I've set 3 different states for if the signature is verified:
+// VALID - The token parsed without errors and the signature was successfully validated.
+// INVALID - The signature attempted to validate with the matched public key, but it was a bad key.
+// UNKNOWN - The default state. This can mean that no key id matched, or simply no keys were provided to validate against.
+const (
+	Valid   = "client_valid"
+	Invalid = "client_invalid"
+	Unknown = "client_unverified"
+)
+
+// Values for include_raw_jwt column.
+var (
+	allowedIncludeValues = []string{"true", "false", "1", "0"}
+)
+
+// Created errors here to handle switching the verified value depending on the returned error.
+var (
+	ErrMissingKeyId     = errors.New("no key id found in the JWT header")
+	ErrMatchingKeyId    = errors.New("no key id matched the JWT header key id")
+	ErrParsingPemBlock  = errors.New("error parsing PEM block containing the public key")
+	ErrParsingPublicKey = errors.New("error parsing the public key from the PEM block")
+)
+
+type Table struct {
+	slogger *slog.Logger
+}
+
+func TablePlugin(slogger *slog.Logger) *table.Plugin {
+	columns := dataflattentable.Columns(
+		table.TextColumn("path"),
+		table.TextColumn("signing_keys"),
+		table.TextColumn("include_raw_jwt"),
+	)
+
+	t := &Table{
+		slogger: slogger.With("table", "kolide_jwt"),
+	}
+
+	return table.NewPlugin("kolide_jwt", columns, t.generate)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	paths := tablehelpers.GetConstraints(queryContext, "path")
+	if len(paths) < 1 {
+		return nil, fmt.Errorf("kolide_jwt requires at least one path to be specified")
+	}
+
+	for _, path := range paths {
+		for _, keyJSON := range tablehelpers.GetConstraints(queryContext, "signing_keys", tablehelpers.WithDefaults("")) {
+			for _, includeRawJWT := range tablehelpers.GetConstraints(queryContext, "include_raw_jwt", tablehelpers.WithAllowedValues(allowedIncludeValues), tablehelpers.WithDefaults("false")) {
+				for _, dataQuery := range tablehelpers.GetConstraints(queryContext, "query", tablehelpers.WithDefaults("*")) {
+					rawData, err := os.ReadFile(path)
+					if err != nil {
+						t.slogger.Log(ctx, slog.LevelInfo, "error reading JWT data file", "err", err)
+						continue
+					}
+
+					// Parse provided JWT signing keys into an map for verification in the JWT parser
+					var keyMap map[string]string
+					if err := json.Unmarshal([]byte(keyJSON), &keyMap); err != nil {
+						t.slogger.Log(ctx, slog.LevelInfo, "error unmarshaling JWT signing keys", "err", err)
+					}
+
+					row := map[string]interface{}{"verified": Unknown}
```

Comment:
Oh! I see why I was confused about column names. This should not be called `row` it's not a returned row of results. Maybe `data` or something.

Reply:
`data` it is!